srcpkgs/libircclient/template

```diff
@@ -0,0 +1,54 @@
+# Template file for 'libircclient'
+pkgname=libircclient
+version=1.10
+revision=1
+build_style=gnu-configure
+configure_args="--enable-openssl --enable-ipv6 --enable-shared"
+makedepends="libressl-devel"
+short_desc="IRC client library"
+maintainer="maciozo@maciozo.com <maciozo@maciozo.com>"
+license="LGPL-3.0-or-later"
+homepage="https://www.ulduzsoft.com/linux/libircclient/"
+distfiles="https://netix.dl.sourceforge.net/project/libircclient/libircclient/${version}/libircclient-${version}.tar.gz"
+checksum=bbb26f3af348b252c5204917a7f91cfdf172f1b6afbf4df1e561b03e20503c2d
+
+case "$XBPS_TARGET_MACHINE" in
+	*-musl)	depends+=" musl-devel";;
+	*)		depends+=" glibc-devel";;
+esac
+
+do_install() {
+	mkdir -p ${PKGDESTDIR}/usr/lib
+	mkdir -p ${PKGDESTDIR}/usr/include
+	mkdir -p ${PKGDESTDIR}/doc/sources
+	cp ${wrksrc}/src/*.so ${PKGDESTDIR}/usr/lib/
+	cp ${wrksrc}/src/*.a ${PKGDESTDIR}/usr/lib/
+
+	for _sharedlib in ${PKGDESTDIR}/usr/lib/*.so; do
+		echo $(basename ${_sharedlib})
+		_soname=$(objdump -p ${_sharedlib} | awk '/SONAME/ {print $2}')
+		ln -s $(basename ${_sharedlib}) ${PKGDESTDIR}/usr/lib/${_soname}
+	done
+
+	cp ${wrksrc}/include/* ${PKGDESTDIR}/usr/include/
+	cp ${wrksrc}/doc/*.rst ${PKGDESTDIR}/doc/
+	cp ${wrksrc}/doc/sources/*.rst ${PKGDESTDIR}/doc/sources/
+}
+
+libircclient-devel_package() {
+	depends="${sourcepkg}>=${version}_${revision}"
+	short_desc+=" - development files"
+
+	pkg_install() {
+		mkdir -p ${PKGDESTDIR}/usr/share/doc/${pkgname}
+		mv ${DESTDIR}/doc/* ${PKGDESTDIR}/usr/share/doc/${pkgname}
+
+		mkdir -p ${PKGDESTDIR}/usr/lib/pkgconfig
+		cp ${FILESDIR}/libircclient.pc ${PKGDESTDIR}/usr/lib/pkgconfig/
+
+
+		vmove usr/include
+		vmove "usr/lib/*.a"
+		vman man/libircclient.1
```

Comment:
Also `vmove "usr/lib/*.so"` because these will be used by packages depending on this one and are generally not required in the `libircclient` package. There only the versioned `*.so.*` files should reside.

Reply:
Won't this break the symlink `libircclient.so.x -> libircclient.so` for users who don't install the devel package?

Reply:
No, it's the other way round, or vice-versa: `libircclient.so` is a symbolic link to `libircclient.so.x`. Packages linking against `libircclient.so` will thus resolve that symlink to the numbered so.
And `common/shlibs` is Void's way to resolve a dependency from a numbered so back to the library (and version), which then saves you from explicitly defining the required libraries in `depends=…`.

Reply:
Does that mean I have to rename `libircclient.so` to `libircclient.so.x` in the `do_install()` of `libircclient`?

Reply:
Just so we're clear, the makefile produces `libircclient.so`, not `libircclient.so.x`. My experience with libraries is minimal, so I don't know if this is how it's supposed to be done. If not, I can just patch it to produce `libircclient.so.x` instead.